staging/src/k8s.io/sample-controller/controller.go

```diff
@@ -219,7 +219,9 @@ func (c *Controller) processNextWorkItem() bool {
 		// Run the syncHandler, passing it the namespace/name string of the
 		// Foo resource to be synced.
 		if err := c.syncHandler(key); err != nil {
-			return fmt.Errorf("error syncing '%s': %s", key, err.Error())
+			// Put the item back on the workqueue to handle any transient errors.
```

Comment:
could you check whether the error is transient before adding it back to the queue?

Reply:
I'm not sure how the error checking would work from here. Its currently up to the syncHandler to decide which errors are transient and which aren't. If an error is transient then the syncHandler method returns that error as it does [here](https://github.com/kubernetes/kubernetes/blob/8f86654b8e23bbf960a286712df35c003f7389f8/staging/src/k8s.io/sample-controller/controller.go#L304-L309). If its a non-transient error then the syncHandler calls `runtime.HandleError(...)` and returns nil as it does [here](https://github.com/kubernetes/kubernetes/blob/8f86654b8e23bbf960a286712df35c003f7389f8/staging/src/k8s.io/sample-controller/controller.go#L266-L272).

This change just ensures that the transient errors are actually put back on the queue.